core/src/test/java/kafka/server/share/SharePartitionManagerTest.java

```diff
@@ -2804,6 +2805,39 @@ public void testFetchMessagesRotatePartitions() {
         validateRotatedListEquals(topicIdPartitions, resultShareFetch.topicIdPartitions(), 1);
     }
 
+    @Test
+    public void testCreateIdleShareFetchTask() throws Exception {
+        ReplicaManager replicaManager = mock(ReplicaManager.class);
+
+        MockTimer mockTimer = new MockTimer((MockTime) time);
```

Comment:
Hmmm, instead of casting should we move the `time` in the class to `MockTime` itself?